MaxText/inference_microbenchmark_sweep.py

```diff
@@ -0,0 +1,154 @@
+import jsonlines
+import inference_microbenchmark
+import max_utils
+import pyconfig
+from jax._src.lib import xla_extension
+
+
+def main():
+  """
+  User needs to set the config's inference_metadata_file, which is a path to a
+  json file.
+
+  This json should contain the following keys:
+    - key_value_axis_order_product_id_list: comma separated string of key_value_axis_order_product_id
+    - prefill_key_axis_order_list: comma delimited string of prefill_key_axis_order
+    - prefill_value_axis_order_list: comma delimited string of prefill_value_axis_order
+    - ar_key_axis_order_list: comma delimited string of ar_key_axis_order
+    - ar_value_axis_order_list: comma delimited string of ar_value_axis_order
+    - accelerator: name of the accelerator
+    - flatten_microbenchmark_results: Whether or not to flatten results. Should
+      be true
+  """
+  pyconfig.initialize(sys.argv)
+  config = pyconfig.config
+
+  with open(config.inference_metadata_file, encoding='utf-8') as json_file:
+    inference_metadata = json.load(json_file)
+    print(f"inference_metadata: {inference_metadata}")
+
+  key_value_axis_order_product_id_list = inference_metadata['key_value_axis_order_product_id_list'].split(':')
+  prefill_key_axis_order_list = inference_metadata['prefill_key_axis_order_list'].split(':')
+  prefill_value_axis_order_list = inference_metadata['prefill_value_axis_order_list'].split(':')
+  ar_key_axis_order_list = inference_metadata['ar_key_axis_order_list'].split(':')
+  ar_value_axis_order_list = inference_metadata['ar_value_axis_order_list'].split(':')
+
+  results = []
+  for (
+    key_value_axis_order_product_id,
+    prefill_key_axis_order,
+    prefill_value_axis_order,
+    ar_key_axis_order,
+    ar_value_axis_order,
+  ) in zip(
+    key_value_axis_order_product_id_list,
+    prefill_key_axis_order_list,
+    prefill_value_axis_order_list,
+    ar_key_axis_order_list,
+    ar_value_axis_order_list,
+  ):
+    print(f"key_value_axis_order_product_id {key_value_axis_order_product_id}")
+    print(f"prefill_key_axis_order {prefill_key_axis_order}")
+    print(f"prefill_value_axis_order {prefill_value_axis_order}")
+    print(f"ar_key_axis_order {ar_key_axis_order}")
+    print(f"ar_value_axis_order {ar_value_axis_order}")
+
+    # Manually update the config
+    # Don't set key_value_axis_order_product_id; otherwise it will recompute
+    # ar_key_axis_order and ar_value_axis_order
+    quant = 'bf16' if not config.quantization else config.quantization
+    run_name = (
+      f"{inference_metadata['accelerator']}-{config.model_name}-"
+      f"{quant}-{key_value_axis_order_product_id}-{prefill_key_axis_order}-"
+      f"{ar_key_axis_order}"
+    )
+    tensorboard_dir = os.path.join(config.base_output_directory, run_name, "tensorboard", "")
+    checkpoint_dir = os.path.join(config.base_output_directory, run_name, "checkpoint", "")
+    metrics_dir = os.path.join(config.base_output_directory, run_name, "metrics", "")
```

Comment:
```suggestion
    # Manually update the config
    # Don't set key_value_axis_order_product_id; otherwise it will recompute
    # ar_key_axis_order and ar_value_axis_order
    quant = 'bf16' if not config.quantization else config.quantization
    run_name = (
      f"{inference_metadata['accelerator']}-{config.model_name}-"
      f"{quant}-{key_value_axis_order_product_id}-{prefill_key_axis_order}-"
      f"{ar_key_axis_order}"
    )
    tensorboard_dir = os.path.join(config.base_output_directory, run_name, "tensorboard", "")
    checkpoint_dir = os.path.join(config.base_output_directory, run_name, "checkpoint", "")
    metrics_dir = os.path.join(config.base_output_directory, run_name, "metrics", "")
```
```suggestion
    # Manually update the config
    # Don't set key_value_axis_order_product_id; otherwise it will recompute
    # ar_key_axis_order and ar_value_axis_order
    run_tag = (
      f"{key_value_axis_order_product_id}-{prefill_key_axis_order.replace(',','')}-{ar_key_axis_order.replace(',','')}"
    )
    run_name = f"{base_run_name}/{run_tag}"

    tensorboard_dir = os.path.join(config.base_output_directory, run_name, "tensorboard")```

Reply:
@morgandu 

I think if we set `run_name = f"{base_run_name}/{run_tag}"` then the GCS bucket will look like `inference-benchmarks/logs/llama2-7b/microbenchmark/int8/0:1:2:3:4:5:6:7:8:9:10:11:12:13:14:15:16:17:18:19:20:21:22:23:24:25:26:27:28:29:30:31:32:33:34:35:36:37:38:39:40:41:42:43:44:45:46:47/0-0123-0123`

This is because the original `run_name` is set to the the list of product_ids https://github.com/GoogleCloudPlatform/ml-auto-solutions/pull/288/files#diff-0ca160e04db7871a01e5f1aafc9705f3f52efe9b1321b2901289554e806f55d9R90

I feel like having `0:1:2:3:4:5:6:7:8:9:10:11:12:13:14:15:16:17:18:19:20:21:22:23:24:25:26:27:28:29:30:31:32:33:34:35:36:37:38:39:40:41:42:43:44:45:46:47` (which is all the product ids for the single VM) is not necessary, and would only lead to confusion since that's not important to the end consumer of the metrics. Unless you think it's helpful to assign logging per VM?

Is this what you have in mind with this change? If not, we can remove the base_run_name